### PR TITLE
jsx to tsx conversion

### PR DESCRIPTION
**Pull request by mattfredericksen**

### CODE REVIEW

File: frontend/.eslintrc.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ const sharedRules = {
   "@typescript-eslint/no-explicit-any": "off",
   "react-hooks/exhaustive-deps": "off", // TODO: remove this line
   "react-hooks/rules-of-hooks": "off", // TODO: remove this line
+  "@typescript-eslint/explicit-module-boundary-types": "off",
```
Comment (mattfredericksen) [inline frontend/.eslintrc.js:11]: This rule doesn't seem helpful for us and is annoying to comply with. Lmk if anyone wants more justification for disabling it.

File: frontend/pages/_app.tsx [hunk 2]
```diff
@@ -55,14 +55,8 @@ const App: NextPage<AppProps> = ({ Component, pageProps, err }: ExtendedAppProps
...
     }
+  }, []);
 
-    // Scroll to the top of the page whenever router.push() is used.
-    // (The next/Link component automatically handles page scrolling,
-    // but router.push() does not.)
-    const handle = () => window.scrollTo({ top: 0, left: 0, behavior: "smooth" });
-    router.events.on("routeChangeComplete", handle);
-    return () => router.events.off("routeChangeComplete", handle);
-  }, [router.events]);
```
Comment (mattfredericksen) [inline frontend/pages/_app.tsx:65]: Unrelated to the conversion, but no longer needed.
https://nextjs.org/blog/next-10-1#router-methods-scroll-to-top

File: frontend/components/search/SearchButton.tsx [hunk 3]
```diff
@@ -1,11 +1,9 @@
-import { Button, CircularProgress } from "@material-ui/core";
+import { Button, ButtonProps, CircularProgress } from "@material-ui/core";
 import { useRouter } from "next/router";
 import { FC, useContext, useEffect } from "react";
 import PageTransitionContext from "../PageTransitionContext";
 
-interface SearchButtonProps {
-  onClick: () => Promise<boolean>;
-}
+type SearchButtonProps = ButtonProps & { onClick: ButtonProps["onClick"] };
```
Comment (mattfredericksen) [inline frontend/components/search/SearchButton.tsx:6]: This just makes the `onClick` prop required.

File: frontend/.eslintrc.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ const sharedRules = {
   "@typescript-eslint/no-explicit-any": "off",
   "react-hooks/exhaustive-deps": "off", // TODO: remove this line
```
Comment (iacobfred) [inline frontend/.eslintrc.js:9]: Can we go ahead and turn the react-hooks rules back on (by removing lines 9-10)?

**Review by iacobfred — APPROVED** [review]
LGTM

**Pull request merged**